### DISCUSSION TITLE
Module 'mypackage' has no attribute 'Map'

### DISCUSSION THREAD

**Discussion by phillipslucas** [giswqs/geohey#16] [Q&A] [answered]
Hey Everyone,

Repo in question:
[https://github.com/phillipslucas/TheBeans](https://github.com/phillipslucas/TheBeans)

I have successfully created a map class, updated my version and created a new release on pypi with the added functionality to thebeans.py without errors.

When trying to test the map class through my new package in ExampleBeans.ipynb I can install my package (thebeans) but it doesn't recognize Map as an attribute of the package. Likewise when I revert to the geohey repo and try to run the same process I get the same error. 

I think I'm just missing a key piece of information to make this run properly but I'm having a hard time piecing it together myself.

I'd appreciate any and all help,
Lucas

**Comment by phillipslucas** [comment]
From a solid day of troubleshooting I am quite lost. Stackexchange threads suggest the way I'm importing the modules could be circular but I can't find the actual.... circle part of it. My package will install but no Map attribute can be found.

![image](https://github.com/giswqs/geohey/assets/16848326/54f587df-91f1-44d7-bd5e-5c0e9c7274f7)


Now when I try and run the geohey repo examples I get similar errors. 
![image](https://github.com/giswqs/geohey/assets/16848326/2e0182c2-b75f-44e5-83db-6eef86c12d03)

Does anyone have anything similar happening?

**Comment by giswqs** [comment, marked as answer]
Did you install the package into your conda env using `pip install -e .`?

**Comment by giswqs** [comment]
From my testing, the `Map` class exists, but you need to change `ipyleaflet.LayerControl()` to `ipyleaflet.LayersControl()`

![image](https://github.com/giswqs/geohey/assets/5016453/7574d35e-cfea-423e-b9ac-fae8f6baa228)

**Comment by phillipslucas** [comment]
`pip install -e . thebeans ` fixed the problem and now my notebook recognizes the class. I fixed '.LayersControl' as well and everything is up and running.

I missed the `-e .` install notes and its not on the course website. Do you recall what lecture it was addressed in?

**Reply by giswqs** [reply]
I forgot which lecture covers this, but we at least mentioned two to three times in class
